src/feed-provider-lookup.c

```diff
@@ -448,14 +448,32 @@ append_providers_in_directory_to_ptr_array (GFile                *directory,
   return TRUE;
 }
 
+static GStrv
+get_force_additional_languages_from_gsettings (void)
+{
+  GSettingsSchemaSource *schema_source = g_settings_schema_source_get_default ();
+  g_autoptr(GSettingsSchema) schema = g_settings_schema_source_lookup (schema_source,
+                                                                       "com.endlessm.DiscoveryFeed",
+                                                                       TRUE);
+  const gchar *empty[] = {
+    NULL
+  };
+
+  if (schema != NULL)
+    {
+      g_autoptr(GSettings) settings = g_settings_new_full (schema, NULL, NULL);
+      return g_settings_get_strv (settings, "force-additional-languages");
+    }
+
+  return g_strdupv ((GStrv) empty);
```

Comment:
`g_strdupv()` will just return `NULL` if passed `NULL`, so there should be no need for this, you can just return `NULL`

Reply:
Will that work? We dereference the strv during the iteration process to check if we are at the sentinel element:

```
  for (iter = (const gchar * const *) force_additional_languages; *iter != NULL; ++iter)
    g_ptr_array_add (languages, g_strdup (*iter));
```